internal/events/event_poller.go

```diff
@@ -165,7 +166,7 @@ func (ep *eventPoller) readPage() ([]fftypes.LocallySequenced, error) {
 	// a rewind based on it.
 	rewind, pollingOffset := ep.conf.maybeRewind()
 	if rewind {
-		ep.rewindPollingOffset(pollingOffset)
+		pollingOffset = ep.rewindPollingOffset(pollingOffset)
```

Comment:
We had a check in the `rewindPollingOffset` function that we only go backwards in time.

However we didn't return the choice we made - so the calling function was rewinding forwards. This cause a really hard to track down bug that's eaten a bunch of my cycles.

The reason we're only seeing this now, is because with the batches for broadcast and private both rewinding, it's much more likely you'll hit a rewind forwards scenario.